Extend config section, Closes #259

Tschakki opened the pull request:
Closes #259

ishantiw commented on user-guide.md:
Could be improved,
Possible values for `<NETWORK>` are, `test` for `testnet, `main` for `mainnet`, `alpha` for `alphanet` and  `beta` for `betanet`

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
To describe important steps in this innovation, websites, and apps are generally categorized into different Web "versions":
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
    Web 1.0 is the first "version" of the Web, i.e., the first generation of websites that occurred when the Internet became available to the broader public around 1990.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
    However, some Web 2.0 capabilities were present in the days of Web 1.0 but were implemented differently.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
Concerning frontend app development, there aren't many differences between Web2 and Web3 apps when it comes to programming languages.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
To support developers, many blockchain platforms offer their own SDKs and tools to interact with their blockchain, which can make development easier and more efficient.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
In Web3 apps, some data is stored on the blockchain, i.e., a decentralized database.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
How the data is stored is defined inside of smart contracts.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
To interact with the blockchain, there are various libraries and frameworks available, like [viem](https://viem.sh/), [web3.js](https://web3js.readthedocs.io), or [ethers.js](https://docs.ethers.io).
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
To get started with the actual development of a Web3 app, you need to be clear about the right **platform(s)** to build your Web3 app on.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
Additionally, they don't need to be installed by the user and can be used directly in the browser.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
A Progressive Web App (PWA) allows users to install a web app on their devices.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
Additionally, PWAs allow a user to directly install the app through your web app without the need to visit an app store.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
[^1]: Only supported in PWA-compatible browsers (most Chromium-based browsers should work).
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
    Web 2.0 is the era where social media, [web applications](https://en.wikipedia.org/wiki/Web_application), and mobile apps came to the rise.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
    Some of the most popular examples of Web 2.0 applications are Facebook, X, Amazon, and YouTube.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
    Many of these applications and services became important pillars of the internet and people's lives, and even society as a whole.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
    However, the more relevant these applications became, the more obvious and relevant their limitations became, as users had to trust the companies behind them with their data and privacy.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
    Though Web 2.0 revolutionized how people positively used the web, a few of its characteristics led to certain disadvantages for users interacting with Web2 applications. For example:
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
    Web3 is based on the idea of a decentralized internet, where users are in control over their data.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
    2015 marked the beginning of the Web3 era when Ethereum launched as the world's first programmable blockchain.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
    Often, Web3 apps combine other cutting-edge technologies like AI to provide unique and new user experiences.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
      - Decentralized crowdfunding and investing platforms
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
    Web3 promises to bring back the original vision of the internet, where users are in control of their data and privacy and where the internet is a place for free speech and innovation.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
- [Video tutorial: Build a Web3 PWA](https://www.youtube.com/watch?v=VU8i-dn2_GE) *by thirdweb*
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
They appear as native apps on desktop or mobile devices and offer features like offline support and push notifications.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
This way, PWAs allow the development of an application that can be used via a browser[^1], mobile, or desktop application alike.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
Therefore, PWAs are a great choice for reaching users on different platforms simultaneously.
```

TalhaMaliktz reviewed (changes requested):
Please consider the following!

TalhaMaliktz reviewed (approved):
LGTM 👍🏼

TalhaMaliktz reviewed (changes requested):
Broken anchor found!
<img width="1018" alt="image" src="https://github.com/user-attachments/assets/d00fe16c-2553-438e-8794-5ed59d429cf6" />

Tschakki commented:
> Broken anchor found! <img alt="image" width="1018" src="https://private-user-images.githubusercontent.com/13951043/401963303-d00fe16c-2553-438e-8794-5ed59d429cf6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY1MDc4NjIsIm5iZiI6MTczNjUwNzU2MiwicGF0aCI6Ii8xMzk1MTA0My80MDE5NjMzMDMtZDAwZmUxNmMtMjU1My00MzhlLTg3OTQtNWVkNTlkNDI5Y2Y2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEwVDExMTI0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3MGQ5NTczYTFjMWE5NTI0YTI2NzU2MTgxOWJiM2I0ODk1ODAwNmMzYjYyMWE1MTM4ZWM1Y2Q4YzAzODdhODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.pC_wQI27b2WsBF8fB3kyrFEqzVIS0qu6oSADZgji374">

this seems to be in the translation `/ind/`, which doesn't exist yet for this content... How did you get the error?

TalhaMaliktz commented:
> > Broken anchor found! <img alt="image" width="1018" src="https://private-user-images.githubusercontent.com/13951043/401963303-d00fe16c-2553-438e-8794-5ed59d429cf6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY1MDc4NjIsIm5iZiI6MTczNjUwNzU2MiwicGF0aCI6Ii8xMzk1MTA0My80MDE5NjMzMDMtZDAwZmUxNmMtMjU1My00MzhlLTg3OTQtNWVkNTlkNDI5Y2Y2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEwVDExMTI0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3MGQ5NTczYTFjMWE5NTI0YTI2NzU2MTgxOWJiM2I0ODk1ODAwNmMzYjYyMWE1MTM4ZWM1Y2Q4YzAzODdhODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.pC_wQI27b2WsBF8fB3kyrFEqzVIS0qu6oSADZgji374">
> 
> this seems to be in the translation `/ind/`, which doesn't exist yet for this content... How did you get the error?

I got it when I ran `npm run build && npm run serve`; if it is missing a file, I think we can make it part of our workflow to add a file in the `/ind` folder as well whenever we have the final and approved English version of the new file.

matjazv reviewed (approved):
Current documentation looks good to me!

Maybe consider extending it with below topics if you find them useful for this page:
- Security considerations in Web3 applications (smart contract audits, secure key management, handling user data, bug bounties)
- Testing and deployment (local development env, testnets, continuous integration, monitoring)
- Onboarding Web2 users (abstracting wallet setup, gas fee management, support and community)

Tschakki commented:
> > > Broken anchor found! <img alt="image" width="1018" src="https://private-user-images.githubusercontent.com/13951043/401963303-d00fe16c-2553-438e-8794-5ed59d429cf6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY1MDc4NjIsIm5iZiI6MTczNjUwNzU2MiwicGF0aCI6Ii8xMzk1MTA0My80MDE5NjMzMDMtZDAwZmUxNmMtMjU1My00MzhlLTg3OTQtNWVkNTlkNDI5Y2Y2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEwVDExMTI0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3MGQ5NTczYTFjMWE5NTI0YTI2NzU2MTgxOWJiM2I0ODk1ODAwNmMzYjYyMWE1MTM4ZWM1Y2Q4YzAzODdhODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.pC_wQI27b2WsBF8fB3kyrFEqzVIS0qu6oSADZgji374">
> > 
> > 
> > this seems to be in the translation `/ind/`, which doesn't exist yet for this content... How did you get the error?
> 
> I got it when I ran `npm run build && npm run serve`; if it is missing a file, I think we can make it part of our workflow to add a file in the `/ind` folder as well whenever we have the final and approved English version of the new file.

I added the file, which did not resolve the error. Then I changed format of the heading, and it fixed it, so I changed it in the original file as well.. not sure why the error was showing up for the translation and not original file though 🤔

TalhaMaliktz reviewed (approved):
LGTM! 👍🏼

matjazv commented on docs/building-on-lisk/web3-app-development.mdx:
Perhaps add a few words regarding data privacy and visibility on-chain.
Any data on a public blockchain is also viewable by anyone. Developers need to be aware that:
- Once data is added to a public blockchain, it becomes virtually impossible to remove and is visible to everyone.
- Sensitive or personal data may require additional measures (for example, encryption or zero-knowledge techniques) to avoid unwanted exposure.

matjazv commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
In addition, storing large amounts of data onchain is both expensive and slow.
```

matjazv reviewed (approved):
LGTM!
Nice work!

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
### Further Reading
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
A significant challenge for Web3 applications is delivering a user experience that is as seamless as what users expect from Web2 applications.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
To onboard Web2 users to Web3, apps need to simplify user interaction so that it feels natural and intuitive to them while still providing the benefits of Web3 in a secure way.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
To aim for a smooth user experience, and to abstract the complexity of blockchain technology away from the user, it is highly recommended to include [Account Abstraction](https://www.alchemy.com/overviews/what-is-account-abstraction) (sometimes also referred to as "smart accounts", "smart wallets" or "smart contract accounts") into your Web3 application.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
- **UserOperation**: A structure that describes a transaction to be sent on behalf of a user.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
Like a transaction, it contains `sender`, `to`, `calldata`, `maxFeePerGas`, `maxPriorityFee`, `signature`, `nonce`.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
Unlike a transaction, the use of the `signature` field is not defined by the protocol but by each account implementation.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
- **Account Contract**: The user's smart account sending a user operation.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
- **EntryPoint**: A singleton contract to validate and execute bundles of UserOperations.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
- [Deploying a smart contract with Foundry](/building-on-lisk/deploying-smart-contract/with-Foundry#testing-the-smart-contract)
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
The links should be replaced with guides about testing with Hardhat and Remix.

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
Should this be removed?
```suggestion
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
Virtual Testnets are easy to set up and configure, giving you full control and flexibility over your environments.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
To deal with stuff like wallets, and other Web3-specific things, you might want to use specialized Web3 E2E testing tools.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
Already abbreviated
```suggestion
[Synpress](https://synpress.io/) is an E2E testing library for Web3 dapps. 
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
Synpress can be used with two leading E2E testing frameworks out there, i.e., [Playwright](https://playwright.dev/) and [Cypress](https://www.cypress.io/), with support for all of their features.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
There are also SDKs for Web3 app development, like [thirdweb](https://thirdweb.com/) or [Alchemy](https://www.alchemy.com/), which take away most of the complexity of blockchain development and provide a smooth developer experience.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
Especially sensitive and private user data should not be stored on the blockchain, except if it is encrypted.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
Hardhat allows you to deploy your contracts, [run your tests](https://hardhat.org/tutorial/testing-contracts), and debug your code, all within the confines of your local machine.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
However, it should be noted that onchain data on a public blockchain is visible to everyone and, therefore, not suitable for all types of data.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
The intrinsic vulnerabilities of smart contracts, the immutability of blockchain data, and the interactions with a decentralized network, call for specific testing and security measures.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
Gladly, many tools are already available to help developers ensure the security and reliability of their Web3 apps.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
There are several tools available that help developers analyze their smart contracts for common security issues and vulnerabilities, for example:
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
This specifically includes testing the frontend of your application and ensuring that the user interface works as expected.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
Measure and improve your app's performance, throughput, and resilience under different levels of traffic and transactions.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
I would wait for confirmation from marketing before choosing a capitalization style.

TalhaMaliktz reviewed (changes requested):
Looking good already!

Tschakki commented on docs/building-on-lisk/web3-app-development.mdx:
I wouldn't add a link to an external site here, as we explain below account abstraction ourselves to the reader. We could add it under the further reading section, if you consider this resource to be valuable.

Tschakki commented on docs/building-on-lisk/web3-app-development.mdx:
```suggestion
This specifically includes testing the frontend of your application, which ensures that the user interface works as expected.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
Sure, the further reading section sounds suitable as well! I would use this [Blog](https://www.alchemy.com/blog/account-abstraction), which I found very useful for understanding AA.

Tschakki commented on docs/building-on-lisk/web3-app-development.mdx:
Yes, we should extend the guides with examples how to test the smart contract, I created the corresponding issues now 👍

Tschakki commented on docs/building-on-lisk/web3-app-development.mdx:
added the link below!

Tschakki commented on docs/building-on-lisk/web3-app-development.mdx:
I updated the headings now to be all sentence case 👍

TalhaMaliktz reviewed (approved):
LGTM! 👍🏼
